Updog: use compile time specific json payloads in unit tests

jpmcb opened the pull request:
**Issue number:**

Closes #2366

**Description of changes:**

```
Modifies the updog unit tests to use architecture specific json stub
files when those tests make assertions based on the host architecture.

This patch fixes unit tests not working on aarch machines due to hard
coded "x86_64" string literals.

Signed-off-by: John McBride <jpmmcb@amazon.com>
```

Adds a few files for `aarch64` based machines and modifies the names of previous files to be have the new naming convention.

**Testing done:**

Now able to fully run `cargo make unit-tests` with no errors

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

etungsten commented on sources/updater/updog/src/main.rs:
```diff
@@ -700,8 +700,9 @@ mod tests {
-        // A manifest with two updates, both less than 0.1.3
-        let path = "tests/data/example_3.json";
+        // A manifest with two updates, both less than 0.1.3.
+        // Use a architecture specific JSON payload, otherwise Updog will ignore the update
```
nit: we use lowercase when referencing our first party binaries.
```suggestion
        // Use a architecture specific JSON payload, otherwise updog will ignore the update
```

This applies to the commit message as well.

jpmcb replied:
Gotcha! Good to know 👍🏼

etungsten reviewed (approved):
:dog2:

jpmcb committed:
updog: use compile time specific json payloads in unit tests

Modifies the updog unit tests to use architecture specific json stub
files when those tests make assertions based on the host architecture.

This patch fixes unit tests not working on aarch machines due to hard
coded "x86_64" string literals.

Signed-off-by: John McBride <jpmmcb@amazon.com>

jpmcb commented:
Force pushed to update `%s/Updog/updog/` in comment and commit message

zmrow reviewed (approved):
:singer:
